### PR TITLE
feat: add auth rout with jwt and password

### PR DESCRIPTION
**Pull request by SlikeX**
Created auth roure via using JWT and passport library.
Questions:
1) About utils folder, is it ok to keep thise files in Utils folder? 
2)Could u give advice about app structure? I mean folder's name and it contains.

### CODE REVIEW

File: routes/auth.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const JwtStrategy = require('passport-jwt').Strategy;
+const ExtractJwt = require('passport-jwt').ExtractJwt;
+const db = require('../database');
+const { jwtCallback } = require('../passport');
+const { adminGuard } = require('../guards');
+
+const router = express.Router();
+
+router.use(passport.initialize());
+router.use(express.json());
```
Comment (devbarden) [inline routes/auth.js]: why is there ?

File: routes/auth.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+router.use(express.urlencoded({ extended: false }));
```
Comment (devbarden) [inline routes/auth.js]: the same

File: utils/database.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const bcrypt = require('bcrypt');
+
+const users = [
```
Comment (devbarden) [inline utils/database.js]: the data shouldn't be store in js files.

Comment (devbarden) [reply utils/database.js]: and also shouldn't be in utils

File: utils/database.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  {
+    id: 3,
+    email: 'test3@mail.ru',
+    password: 'test123',
+    role: 'admin',
+  },
+];
+
+function getUserByEmail(email) {
```
Comment (devbarden) [inline utils/database.js]: it should be at user controller

File: utils/database.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  return users.find((user) => user.email === email);
+};
+
+function loginUser(email, password) {
```
Comment (devbarden) [inline utils/database.js]: the same

**Review by devbarden — COMMENTED** [review]
please clean your code and split each router. you can create controllers folder and move logic to this folder for every entity.. for example: controllers/accountController.js and inside this module you can put all functions like find update, delete and etc. and at router it will be smth like that
`router.get("/", AccountController.findAll)`

File: utils/database.js (hunk 3 above)
Comment (SlikeX) [reply utils/database.js]: I keep it in js file coz it's king of mock data, in next HW i'll remove it and set database

**Comment by devbarden** [comment]
you can merge it if you want)

**Review by devbarden — APPROVED** [review]

**Pull request merged**